model/plugin_integration.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"encoding/json"
+	"io"
+)
+
+const (
+	ScopeMobile = "mobile"
+	ScopeWebApp = "webapp"
+)
+
+type PluginIntegration struct {
+	PluginID string `json:"id"`
+	// Location defines where the client should present this integration
+	Location string `json:"location"`
+	// RequestURL defines the URL the client will reach to perform the integration action
+	RequestURL string `json:"request_url"`
+	// Scope defines which clients should show this integration
+	Scope []string    `json:"scope"`
+	Extra interface{} `json:"extra"`
```

Comment:
Given that every implementation here and all the examples contain `Icon` and `Text`, I'm wondering if we should remove `Extra` and move `Icon` and `Text` into `PluginIntegration`.